azurerm/internal/services/kusto/resource_arm_kusto_eventgrid_data_connection.go

```diff
@@ -0,0 +1,284 @@
+		Delete: resourceArmKustoEventGridDataConnectionDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMKustoDataConnectionName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"cluster_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMKustoClusterName,
+			},
+
+			"database_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMKustoDatabaseName,
+			},
+
+			"storage_account_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"eventhub_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"consumer_group": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateEventHubConsumerName(),
+			},
+
+			"table_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAzureRMKustoEntityName,
+			},
+
+			"mapping_rule_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAzureRMKustoEntityName,
+			},
+
+			"data_format": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(kusto.MULTIJSON),
+					string(kusto.JSON),
+					string(kusto.CSV),
+					string(kusto.TSV),
+					string(kusto.SCSV),
+					string(kusto.SOHSV),
+					string(kusto.PSV),
+					string(kusto.TXT),
+					string(kusto.RAW),
+					string(kusto.SINGLEJSON),
+					string(kusto.AVRO),
+					string(kusto.TSVE),
+				}, false),
+			},
+		},
+	}
+}
+
+func resourceArmKustoEventGridDataConnectionCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Kusto.DataConnectionsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Azure Kusto EventGrid Data Connection creation.")
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	clusterName := d.Get("cluster_name").(string)
+	databaseName := d.Get("database_name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
```

Comment:
Why is this block required? ARM template is idempotent so there's no reason to check if the resource exists.

Reply:
This block is required because Azure tends to upsert any changes regardless of whether it's a new resource or not. This provides challenges for Terraform users as inadvertent changes may get applied on a resource created by something other than Terraform. This check forces a user to first import a resource that has been created outside of Terraform before making any changes on it. It's an easy simple check to prevent users from accidentally making changes to  their infrastructure.